### DISCUSSION TITLE
Traefik proxy | Can this plugin be installed for free?

### DISCUSSION THREAD

**Discussion by dberardo-com** [tomMoulard/fail2ban#61] [Q&A] [answered]
or does it work only for Enterprise versions of Traefik?

Is installation procedure the same?

**Comment by tomMoulard** [comment, marked as answer]
> Can this plugin be installed for free?

yes

> does it work only for Enterprise versions of Traefik?

no

> Is installation procedure the same?

yes

**Reply by dberardo-com** [reply]
thanks a lot for the quick answer. i might try it out soon on a k3s installation. if needed i will post in here

**Reply by dberardo-com** [reply]
ok, i did some tests and the core functionality is working, however it seems like the plugin is not ppicking up the real ip of the incoming requests, but some sort of internal IP that perhaps kube-proxy uses to forward requests to the Pod. Is there any way to make the plugin "see" the real-IP of the incoming requests ?

**Reply by tomMoulard** [reply]
For now, the plugin only looks at the request's RemoteAddr, it might be interesting to look at the `X-Forwarded-For` header like in the [rate limit's ip strategy source criterion](https://doc.traefik.io/traefik/middlewares/http/ratelimit/#sourcecriterionipstrategy).

**Reply by dberardo-com** [reply]
ok, so it seems that exposing traefik via a LoadBalancer with the following option might do the trick: `externalTrafficPolicy: Local` would you agree?

**Reply by tomMoulard** [reply]
idk

**Reply by dberardo-com** [reply]
ok, we might wait for anyone else out there to confirm this theory then

**Reply by dberardo-com** [reply]
ref: https://github.com/traefik/traefik/issues/9174#issuecomment-1380381974

**Comment by dberardo-com** [comment]
also related to the setup/config procedure --> is fail2ban considering 5xx responses also as "failures" ? if so, it is possible to restrict to only 4xx return codes, ignoring 5xx ?

**Reply by dberardo-com** [reply]
let us know if thats possible @tomMoulard  thanks a lot

**Reply by tomMoulard** [reply]
Sure, we can add a mechanism to filter only for a given http code range, but I do not see a real use case. Why do you want to filter out 5XX codes when a potential user could use this as a way to DOS your backend server

**Reply by tomMoulard** [reply]
This could be linked to #40

**Reply by dberardo-com** [reply]
well for such kind of user i would prefer to use a rate limiting rule. i guess this is also possible with fail2ban, isnt it ? otherwise will use the regular built-in rate limiter from traefik

**Reply by tomMoulard** [reply]
I don't think that this plugin is ment to be a rate limiter, and I do not see how you would apply a rate limiter using Traefik on 5XX error codes. Traefik's rate limiting is only based on the number of concurrent requests. It is not based on the status code of the request.

**Reply by dberardo-com** [reply]
well i would apply a rate limiter regardless of the response code, so this should prevent DOS in any case. again, this is only one specific optional behavior that i need to apply on some service that might return 5xx due to some conditions, so will not be the default. thanks for providing the link to the other issue, i will investigate
